implementations/compucell3d/model.py

```diff
@@ -19,15 +19,32 @@ def _impl_MODEL000(**kwargs):
 def _impl_MODEL003(**kwargs):
     lambda_dir = - float(kwargs['lambda_dir'])
     target_angle = float(kwargs['target_angle'])
+    cpm_force_mode = kwargs['cpm_force_mode']
+    cpm_update_direction = kwargs['cpm_update_direction']
+
+    if cpm_force_mode in ['extension', 'contraction']:
+        raise ValueError(f'CC3D does not support the specified force mode: {cpm_force_mode}')
+    if cpm_update_direction == 'source-to-target-norm':
+        raise ValueError(f'CC3D does not support the specified movement term: {cpm_update_direction}')
+
     return [
         pcs.ExternalPotentialPlugin(lambda_x=lambda_dir * cos(target_angle),
-                                    lambda_y=lambda_dir * sin(target_angle))
+                                    lambda_y=lambda_dir * sin(target_angle),
+                                    com_based=cpm_update_direction == 'cell-mass-displacement')
     ]
 
 
 def _impl_MODEL005(**kwargs):
+    cpm_force_mode = kwargs['cpm_force_mode']
+    cpm_update_direction = kwargs['cpm_update_direction']
+
+    if cpm_force_mode in ['extension', 'contraction']:
```

Comment:
shouldn't this be "retraction" rather than "contraction"?

Reply:
Fixed. Good catch!